shares/shares.go

```diff
@@ -90,33 +72,38 @@ func (s Share) IsCompactShare() (bool, error) {
 	return isCompact, nil
 }
 
+// GetSigner returns the signer of the share, if the
+// share is not of type v1 and is not the first share in a sequence
+// it returns nil
+func GetSigner(share Share) []byte {
+	infoByte := share.InfoByte()
+	if infoByte.Version() != ShareVersionOne {
+		return nil
+	}
+	if !infoByte.IsSequenceStart() {
+		return nil
+	}
+	startIndex := namespace.NamespaceSize + ShareInfoBytes + SequenceLenBytes
+	endIndex := startIndex + blob.SignerSize
+	return share.data[startIndex:endIndex]
+}
+
 // SequenceLen returns the sequence length of this *share and optionally an
 // error. It returns 0, nil if this is a continuation share (i.e. doesn't
 // contain a sequence length).
-func (s *Share) SequenceLen() (sequenceLen uint32, err error) {
-	isSequenceStart, err := s.IsSequenceStart()
-	if err != nil {
-		return 0, err
-	}
-	if !isSequenceStart {
-		return 0, nil
+func (s *Share) SequenceLen() uint32 {
+	if !s.IsSequenceStart() {
+		return 0
 	}
 
 	start := namespace.NamespaceSize + ShareInfoBytes
 	end := start + SequenceLenBytes
-	if len(s.data) < end {
```

Comment:
why was this check deleted? I think it's good to have to avoid out of bounds issues